shared/AppInsightsCore/Tests/Selenium/CookieManager.Tests.ts

```diff
@@ -0,0 +1,370 @@
+                } else {
+                    Assert.equal(newValue + "; expires=Thu, 12 Feb 1970 00:00:00 GMT; max-age=" + maxAge + "; path=/", this._testCookies[newKey]);
+                }
+
+                // With existing max-age
+                manager.del(newKey);
+                Assert.equal("", manager.get(newKey));
+                Assert.equal(undefined, this._testCookies[newKey]);
+
+                manager.set(newKey, newValue + "; expires=Thu, 12 Feb 2170 00:00:00 GMT", maxAge);
+                Assert.equal(newValue, manager.get(newKey));
+                if (this.isEmulatingEs3) {
+                    Assert.equal(newValue + "; expires=Thu, 12 Feb 2170 00:00:00 GMT; path=/", this._testCookies[newKey]);
+                } else {
+                    Assert.equal(newValue + "; expires=Thu, 12 Feb 2170 00:00:00 GMT; max-age=" + maxAge + "; path=/", this._testCookies[newKey]);
+                }
+
+                manager.del(newKey);
+                Assert.equal("", manager.get(newKey));
+                Assert.equal(undefined, this._testCookies[newKey]);
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: Set as Disabled",
+            test: () => {
+                let manager = createCookieMgr({
+                    enabled: false,
+                    getCookie: () => { throw "Should not be called" },
+                    setCookie: () => { throw "Should not be called" },
+                    delCookie: () => { throw "Should not be called" }
+                });
+
+                let newKey = "test." + newId();
+                let newValue = newId();
+                manager.set(newKey, newValue);
+                Assert.equal("", manager.get(newKey));
+
+                manager.del(newKey);
+                Assert.equal("", manager.get(newKey));
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: disable cookies using the legacy setting",
+            test: () => {
+
+                let core = new AppInsightsCore();
+                core.initialize({
+                    instrumentationKey: "testiKey",
+                    isCookieUseDisabled: true,
+                    cookieCfg: {
+                        getCookie: () => { throw "Should not be called" },
+                        setCookie: () => { throw "Should not be called" },
+                        delCookie: () => { throw "Should not be called" }
+                    }
+                } as any, [new ChannelPlugin()]);
+
+                let manager = core.getCookieMgr();
+                let newKey = "test." + newId();
+                let newValue = newId();
+                manager.set(newKey, newValue);
+                Assert.equal("", manager.get(newKey));
+
+                manager.del(newKey);
+                Assert.equal("", manager.get(newKey));
+            }
+        });        
+
+        this.testCase({
```

Comment:
nit: may be add one more test with both legacy setting and current setting and check if legacy setting is ignored :)